File: lib/doc-stats.ts
/**
 * 文档统计：字数、字符数、阅读时间
 */
export interface DocStats {
  wordCount: number;
  charCount: number;
  readingTimeMin: number;
}

export function computeStats(content: string): DocStats {
  if (!content.trim()) return { wordCount: 0, charCount: 0, readingTimeMin: 0 };

  // 去掉 markdown 语法标记
  const cleaned = content
    .replace(/^#{1,6}\s+/gm, '')
    .replace(/[*_~`>|[\]()!-]/g, '')
    .replace(/\n+/g, ' ')
    .trim();

  const charCount = cleaned.length;

  // CJK 字符每个算一词，非 CJK 按空格分词
  const cjkChars = cleaned.match(/[\u4e00-\u9fff\u3400-\u4dbf\uf900-\ufaff]/g);
  const cjkCount = cjkChars ? cjkChars.length : 0;

  // 去掉 CJK 字符后按空格分词
  const nonCjk = cleaned.replace(/[\u4e00-\u9fff\u3400-\u4dbf\uf900-\ufaff]/g, ' ').trim();
  const latinWords = nonCjk ? nonCjk.split(/\s+/).filter(Boolean).length : 0;

  const wordCount = cjkCount + latinWords;

  // 中文 ~400字/分钟，英文 ~200词/分钟，取平均
  const readingTimeMin = Math.max(1, Math.ceil(wordCount / 350));

  return { wordCount, charCount, readingTimeMin };
}

File: backend/lib/doc_stats.py
import re
import math


def compute_stats(content: str) -> dict:
    if not content.strip():
        return {"word_count": 0, "char_count": 0, "reading_time_min": 0}

    cleaned = re.sub(r"^#{1,6}\s+", "", content, flags=re.MULTILINE)
    cleaned = re.sub(r"[*_~`>|[\]()!\-]", "", cleaned)
    cleaned = re.sub(r"\n+", " ", cleaned).strip()

    char_count = len(cleaned)

    cjk_chars = re.findall(r"[\u4e00-\u9fff\u3400-\u4dbf\uf900-\ufaff]", cleaned)
    cjk_count = len(cjk_chars)

    non_cjk = re.sub(r"[\u4e00-\u9fff\u3400-\u4dbf\uf900-\ufaff]", " ", cleaned).strip()
    latin_words = len([w for w in non_cjk.split() if w]) if non_cjk else 0

    word_count = cjk_count + latin_words
    reading_time_min = max(1, math.ceil(word_count / 350))

    return {
        "word_count": word_count,
        "char_count": char_count,
        "reading_time_min": reading_time_min,
    }
